### PR TITLE
Add step When I execute sequence of actions: actions

### PR DESCRIPTION
**Pull request by uarlouski**
Not for review yet

### CODE REVIEW

File: vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/ActionSteps.java [hunk 1]
```diff
@@ -16,90 +16,74 @@
...
-     * <li><code>offset</code> the offset to move by (ex.: (10, 10))</li>
+     * <li><code>type</code> is one of web actions: DOUBLE_CLICK, CLICK_AND_HOLD, MOVE_BY_OFFSET,
+     * RELEASE, SEND_KEYS, CLICK</li>
+     * <li><code>argument</code> either text or search attribute or point</li>
      * </ul>
      * @param actions table of actions to execute
      */
     @When("I execute the sequence of actions: $actions")
-    public void executeActionsSequence(List<Action> actions)
+    public void executeActionsSequence(ExamplesTable actions)
```
Comment (valfirst) [inline vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/ActionSteps.java]: we can't change step API: deprecate the old step, introduce a new step with updated syntax

File: vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/ActionSteps.java [hunk 2]
```diff
@@ -16,90 +16,74 @@
...
     {
         Actions actionSequence = new Actions(webDriverProvider.get());
-        for (Action action : actions)
+        actions.getRowsAsParameters().forEach(parameter ->
```
Comment (ikalinin1) [inline vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/ActionSteps.java]: I propose to move all the parsing logic into a converter. ExamplesTable -> Actions

File: vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/model/ActionType.java [hunk 3]
```diff
@@ -16,66 +16,88 @@
...
+    public Class<?> getArgumentType()
     {
-        return coordinatesRequired;
+        return argumentType;
     }
 
-    public void addAction(Actions actions, Optional<WebElement> element)
+    public abstract void addAction(Actions actions, Object argument);
+
+    public <T, U extends WebElement> void performOnWebelement(T argument, Consumer<U> argumentConsumer)
```
Comment (ikalinin1) [inline vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/model/ActionType.java]: `performOnWebelement` -> `performOnWebElement`

File: vividus-plugin-web-app/src/test/java/org/vividus/bdd/converter/StringToListSequenceActionConverterTests.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    @InjectMocks
+    private StringToListSequenceActionConverter converter;
+
+    @Test
+    void testConvertValueSearchAttributes()
+    {
+        String by = "By.caseSensitiveText(" + TEXT + ")";
+        SearchAttributes searchAttributes = mock(SearchAttributes.class);
+        when(stringToSearchAttributesConverter.convertValue(by, null)).thenReturn(searchAttributes);
+        String value = HEADERS
```
[the remark this answers is not in the document]
Comment (uarlouski) [reply vividus-plugin-web-app/src/test/java/org/vividus/bdd/converter/StringToListSequenceActionConverterTests.java]: this is like copying test data and simplicity of changing the test data vs test readability, I've moved concatenation to separate method to smooth concatenation  sequence out

File: vividus-plugin-web-app/src/test/java/org/vividus/bdd/converter/StringToListSequenceActionConverterTests.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+    @Test
+    void testConvertValueSearchAttributes()
+    {
+        String by = "By.caseSensitiveText(" + TEXT + ")";
+        SearchAttributes searchAttributes = mock(SearchAttributes.class);
+        when(stringToSearchAttributesConverter.convertValue(by, null)).thenReturn(searchAttributes);
+        String value = "|type|argument|\n"
+                     + "|CLICK|By.caseSensitiveText(text)|\n"
+                     + "|DOUBLE_CLICK|By.caseSensitiveText(text)|";
```
Comment (valfirst) [inline vividus-plugin-web-app/src/test/java/org/vividus/bdd/converter/StringToListSequenceActionConverterTests.java]: what about formatting in a readable view?

**Pull request merged**